src/main/java/org/medicmobile/webapp/mobile/RequestPermissionDialog.java

```diff
@@ -0,0 +1,65 @@
+package org.medicmobile.webapp.mobile;
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.support.v4.app.ActivityCompat;
+
+import static org.medicmobile.webapp.mobile.MedicLog.error;
+import static org.medicmobile.webapp.mobile.MedicLog.trace;
+
+/**
+ * Shows a confirmation dialog that displays a "prominent" disclosure about how
+ * the user geolocation data is used, asking to confirm whether to allow the app to
+ * access the location or not.
+ *
+ * If the user accepts, a request to the API to access the location is made,
+ * but Android will show another confirmation dialog. If the user decline the first
+ * confirmation, the request to the API is omitted and the decision recorded to avoid
+ * requesting the same next time.
+ */
+public class RequestPermissionDialog {
+
+	private static final String[] LOCATION_PERMISSIONS = { Manifest.permission.ACCESS_FINE_LOCATION };
+
+	/**
+	 * Show the confirmation dialog unless the user has previously denied
+	 * to share the location from this same dialog.
+	 */
+	public static void show(final Activity activity, final int requestCode) {
+		final SettingsStore settings = SettingsStore.in(activity);
+		if (settings.hasUserDeniedGeolocation()) {
+			trace(activity, "RequestPermissionDialog.show() :: " +
+					"user has previously denied to share location");
+			return;
+		}
+		AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+		// TODO Define the final message and localize
+		AlertDialog alert = builder
+			.setTitle("Use your location")
+			.setIcon(android.R.drawable.ic_menu_mylocation)
+			.setMessage("This app collects location data to enable [feature], [feature], ...")
```

Comment:
This should be stored in the [strings.xml](https://github.com/medic/medic-android/blob/master/src/main/res/values/strings.xml) with the other user facing text. This makes it really easy for projects to change the text as required, including changing the language.

This also applies to the title, and button text.

Reply:
Yes it is just to test but I going to move all to configurations and localized, even the button labels.